controllers/profile.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+var bodyParser = require('body-parser');
+var session = require('express-session');
+var bcrypt = require('bcrypt');
+var db = require('../database.js');
+var config = require('../config.js');
+var MySQLStore = require('express-mysql-session')(session);
+var sessionStore = new MySQLStore(config.dbOptions);
+
+function requireAuthentication(req, res, next) {
```

Comment:
Could you move this to the main index file and define it as `app.middleware.requireAuthentication`? This will allow us to re-use the method in the rest of the project.